src/app/shared/components/template/services/instance/template-row.service.ts

```diff
@@ -244,11 +248,12 @@ export class TemplateRowService extends SyncServiceBase {
     preProcessedRow: FlowTypes.TemplateRow,
     isNestedTemplate: boolean
   ) {
-    const { _nested_name, _evalContext } = preProcessedRow;
+    const { _nested_name, _evalContext = {} } = preProcessedRow;
     // Evaluate row variables in context of current local state
-    const evalContext = {
+    // TODO - only extract local variables as needed
+    const evalContext: FlowTypes.TemplateRowEvalContext = {
       ..._evalContext,
-      templateRowMap: this.templateRowMap,
+      local: { ...this.templateRowMapValues, ..._evalContext.local },
```

Comment:
NOTE- this is a feature change which slipped into this PR from the follow-up work. It shouldn't be too significant, just allowing additional local values to be included in dynamic evaluation context. This will be used by data_items to create intermediate variables without the need for uniqueness across the template (each data item row processes with access to its own value for the created local variable)